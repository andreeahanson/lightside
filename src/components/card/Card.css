.favorite {
  background: yellow;
}

.card {
  display: flex;
  flex-direction: column;
  text-align: center;
  width: 90%;
  margin: 10px;
}