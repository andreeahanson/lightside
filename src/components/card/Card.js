import React from 'react';
import './Card.css';

const Card = (props) => {
  console.log(props.results)
  let allCards = props.results.map(card => {
    return (
      <article class="card">
        <h3>{card.name}</h3>
        <p>{card.birth_year}</p>
        <p>{card.gender}</p>
        <p>{card.height}</p>
        <p>{card.hair_color}</p>
        <button className="favorite">Favorite</button> 

      </article>
    )

  })
  return allCards;
}


export default Card;