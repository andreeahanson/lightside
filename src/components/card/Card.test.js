import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';
import samplePerson from '../../sampleDataDirectory/samplePeople';
import sampleVehicles from '../../sampleDataDirectory/sampleVehicles';
import samplePlanets from '../../sampleDataDirectory/samplePlanets';
import sampleFavorites from '../../sampleDataDirectory/sampleFavorites';

describe('Card', () => {
  let wrapper1;
  let wrapper2;
  let wrapper3;
  let toggleFavoriteMock = jest.fn();
  let classNameMock = '.crap';

  beforeEach(() => {
    wrapper1 = shallow(
      <Card
        data={samplePerson.results}
        favorites={sampleFavorites.results}
        toggleFavorite={toggleFavoriteMock}
        className={classNameMock}
        error='error'
      />
    );

    wrapper2 = shallow(
      <Card
        data={sampleVehicles.results}
        favorites={sampleFavorites.results}
        toggleFavorite={toggleFavoriteMock}
        className={classNameMock}
        eerror='error'
      />
    );

    wrapper3 = shallow(
      <Card
        data={samplePlanets.results}
        favorites={sampleFavorites.results}
        toggleFavorite={toggleFavoriteMock}
        className={classNameMock}
        error='error'
      />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper1).toMatchSnapshot();
    expect(wrapper2).toMatchSnapshot();
    expect(wrapper3).toMatchSnapshot();
  });

  it('should match the snapshot when there are no favorites or data items', () => {
    wrapper1 = shallow(
      <Card data={[]} favorites={[]} className={classNameMock} error='error' />
    );

    expect(wrapper1).toMatchSnapshot();
  });

  it('should invoke toggleFavorite on click', () => {
    wrapper1 = shallow(
      <Card
        data={[
          {
            birth_year: '41.9BBY',
            created: '2014-12-10T16:20:44.310000Z',
            edited: '2014-12-20T21:17:50.327000Z',
            eye_color: 'blue',
            films: [
              'https://swapi.co/api/films/5/',
              'https://swapi.co/api/films/4/',
              'https://swapi.co/api/films/6/'
            ],
            gender: 'male',
            hair_color: 'blond',
            height: '188',
            homeworld: 'https://swapi.co/api/planets/1/',
            mass: '84',
            name: 'Anakin Skywalker',
            skin_color: 'fair',
            species: ['https://swapi.co/api/species/1/'],
            starships: [
              'https://swapi.co/api/starships/59/',
              'https://swapi.co/api/starships/65/',
              'https://swapi.co/api/starships/39/'
            ],
            url: 'https://swapi.co/api/people/11/',
            vehicles: [
              'https://swapi.co/api/vehicles/44/',
              'https://swapi.co/api/vehicles/46/'
            ]
          }
        ]}
        favorites={[]}
        className={classNameMock}
        toggleFavorite={toggleFavoriteMock}
        error='error'
      />
    );

    wrapper1.find('article').simulate('click');

    expect(toggleFavoriteMock).toBeCalled();
  });
});
