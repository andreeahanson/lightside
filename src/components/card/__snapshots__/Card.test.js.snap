// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `ShallowWrapper {}`;

exports[`Card should match the snapshot 2`] = `ShallowWrapper {}`;

exports[`Card should match the snapshot 3`] = `ShallowWrapper {}`;

exports[`Card should match the snapshot when there are no favorites or data items 1`] = `ShallowWrapper {}`;
